synapse/lib/storm.py

```diff
@@ -1104,15 +1104,14 @@ def _stormOperJoin(self, query, oper):
 
         args = oper[1].get('args')
         opts = dict(oper[1].get('kwlist'))
+        if len(args) != 2:
```

Comment:
The `join` operator requires a dst and a src. We should raise an explicit exception if this doesn't happen.

Reply:
This should probably work the same was as pivot ( which assumes the inbound node's primary property is the intended src prop when it's not present )